tests/api-compatibility/README.md

```diff
@@ -0,0 +1,126 @@
+`LastStableApiLevel` - The last stable api level.
+
+`TargetImplementationPath` - Path to the implementation assembly. Based on this path we will generate the contract path by replace current version with previous version
+
+`ApiCompatibilityPath` - Path to the Api Compatibility folder. On this folder we should find the acceptable-breakages files and the reference folder that contians the current Mono.Android.dll assembly.
+
+The *task* is aware of the Api levels and what is the previous Api level for agiven version, so When building api version 2, the Task knows if previous version is the version 1.5 or 1.0 for instance.
+
+## Usage
+
+To use the build task we need to import it to our project. On *Mono.Android.csproj* we add the follow line to refrence the build task to our project:
+
+```<UsingTask AssemblyFile="..\..\bin\Build$(Configuration)\Xamarin.Android.Tools.BootstrapTasks.dll" TaskName="Xamarin.Android.Tools.BootstrapTasks.CheckApiCompatibility" />```
+
+Once task is referenced we can use it by adding the follow lines:
+
+```
+  <PropertyGroup>
+    <ApiCompatibilityDir>$(MSBuildProjectDirectory)/Test/api-compatibility</ApiCompatibilityDir>
+  </PropertyGroup>
+  <ItemGroup>
+    <ApiCompatibilityFiles Include="$(ApiCompatibilityDir)/acceptable-breakages-v*.txt" />
+    <ApiCompatibilityFiles Include="$(ApiCompatibilityDir)/reference/*.*" />
+  </ItemGroup>
+  <Target Name="AfterBuild" Inputs="$(TargetPath);@(ApiCompatibilityFiles)" Outputs="$(IntermediateOutputPath)CheckApiCompatibility.stamp">
+    <CheckApiCompatibility
+        ApiCompatPath="../../packages/Microsoft.DotNet.ApiCompat.5.0.0-beta.19562.5/tools/net472/"
+        ApiLevel="$(AndroidFrameworkVersion)"
+        LastStableApiLevel="$(AndroidLatestStableFrameworkVersion)"
+        TargetImplementationPath="$(OutputPath)"
+        ApiCompatibilityPath = "$(ApiCompatibilityDir)" />
+    <Touch Files="$(IntermediateOutputPath)CheckApiCompatibility.stamp" AlwaysCreate="True" />
+    <ItemGroup>
+      <FileWrites Include="$(IntermediateOutputPath)CheckApiCompatibility.stamp" />
+    </ItemGroup>
+  </Target>
+```
+
+`ApiCompatibilityFiles` - Are the files we would like to monitor for changes, this is doen to prevent the task to run if it is not required. (there were no changes)
+
+`Touch Files` - It is our timestamp file to control if task should run.
+
+[bst]: https://github.com/xamarin/xamarin-android/tree/master/build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks
+
+## ApiCompatibilityPath
+
+On this path we will place the files that contain all acceptable breakages for a given Api version.
+
+for example:
+
+```
+acceptable-breakages-v7.0.txt
+acceptable-breakages-v7.1.txt
+acceptable-breakages-v8.0.txt
+acceptable-breakages-v9.0.txt
+```
+
+`acceptable-breakages-v7.0.txt` - Contains any rules that are accept to be broken in relation to the previous Api level of v7.0 (in this case v6.0)
+
+Api Level v6 used to have thefollow piece of code:
```

Comment:
We need to be careful about "api level" and "target framework version".   Android v6.0, which is `$(TargetFrameworkVersion)`=v6.0, is API level 23: https://source.android.com/setup/start/build-numbers

"Api level v6" is thus easily confusable: does it mean API-6, which is for Android 2.0.1 (2009!), or does it mean TargetFrameworkVersion/Android v6.0, released 2015?

*In general*, "API-" should be for API levels, e.g. API-23, while TargetFrameworkVersion should be used for Android versions.

...which will be completely meaningless come .NET 5, when our TargetFrameworkVersion scheme potentially changes, and we'll need to use some other terminology.  "Android v11.0", perhaps?

Reply:
Updated to use target framework version for now.